### PR TITLE
[ui][run_one] Return a Result<i32> instead of TargetRcUi

### PR DESCRIPTION
**Pull request by chantra**
If an error happen when trying to set up/run the VM, return an Error, otherwise return the return code of the command.

Do not report a non-zero return code from the command as an Error, the command actually ran, but do log the error to the console to make it more obvious to the user.

### CODE REVIEW

File: src/ui.rs [hunk 1]
```diff
@@ -167,16 +160,14 @@ impl Ui {
 
     /// UI for a single target. Must be run on its own thread.
     ///
-    /// Returns a TargetUiRc
-    fn target_ui(updates: Receiver<Output>, target: String, show_cmd: bool) -> TargetUiRc {
+    /// Returns an Error if the vm failed to run the command.
+    /// Otherwise, return the return code of the command.
+    fn target_ui(updates: Receiver<Output>, target: String, show_cmd: bool) -> Result<i32, Error> {
```
Comment (danobi) [inline src/ui.rs:165]: Nit: if you import `anyhow::Result`, you can make it be `-> Result<i32>`

File: src/ui.rs [hunk 2]
```diff
@@ -250,8 +243,10 @@
             term.write_line("FAILED").expect("Failed to write terminal");
         }
 
-        rc.error = errors != 0;
-        rc
+        if errors != 0 {
+            return Err(anyhow!("Failed to run the target"));
```
Comment (danobi) [inline src/ui.rs:247]: Nit: using `anyhow::bail` makes it a little less verbose:

```suggestion
            return bail!("Failed to run the target");
```

**Comment by chantra** [comment]
Thanks @danobi . Nits addressed in the forced-push commit.

**Review by danobi — APPROVED** [review]
thanks!

**Pull request merged**